src/components/extra-attempts-dialog.jsx

```diff
@@ -0,0 +1,289 @@
+
+		// add another extra attempts row
+		const tempAttempts = extraAttempts
+		tempAttempts.set(
+			newIdCount,
+			{
+				id: newIdCount,
+				context_id: '',
+				extra_attempts: 1,
+				user_id: match.id
+			}
+		)
+		setExtraAttempts(tempAttempts)
+		setNewIdCount(newIdCount-1)
+	}
+
+	const onSave = () => {
+		// check to see if course id is null or extra attempts is < 1
+		extraAttempts.forEach((obj) => {
+			if(obj.context_id == '') setSaveError('Must fill in Course ID field')
+		})
+		// post request
+		setExtraAttemptsForInstance(inst.id, Array.from(extraAttempts.values()))
+		.then(onClose())
+	}
+
+	return (
+		<Modal onClose={onClose}>
+			<div className="extraAttemptsModal">
+				<span className="title">Give Students Extra Attempts</span>
+				<div className="attempts-container">
+					<div className="search-container">
+						<span className="search-title">Add students:</span>
+						<input
+							tabIndex="0"
+							value={searchText}
+							onChange={(e) => setSearchText(e.target.value)}
+							type="text"
+							placeholder="Enter a Materia user's name or e-mail"
+							className="attempts-input"/>
+						<div>
+							{ searchResults.length !== 0
+								? <div className="attempts_search_list">
+									{searchResults.map((match) => 
+										<div
+											key={match.id}
+											className='attempts_search_match clickable'
+											onClick={() => onClickMatch(match)}>
+												<img className="attempts_match_avatar" src={match.avatar} />
+												<p className={`attempts_match_name ${match.is_student ? 'attempts_match_student' : ''}`}>{match.first} {match.last}</p>
+										</div>
+									)}
+									</div>
+								: null
+							}
+						</div>
+					</div>
+
+					<div className="attempts_list_container">
+						<div className="headers">
+							<span className="user-header">User</span>
+							<span className="context-header">Course ID</span>
+							<span className="attempts-header">Extra Attempts</span>
+						</div>
+
+						<div className="attempts_list">
+							{Array.from(extraAttempts).map(([attemptId, attemptObj]) => {
+								if(attemptObj.extra_attempts < 0) return
+								const user = users[attemptObj.user_id]
+								if(!user) return <div key={attemptId}>Loading...</div>
+								return <ExtraAttemptsRow
+									key={attemptId}
+									extraAttempt={attemptObj}
+									user={user}
+									onChange={(id, updatedAttempts) => extraAttempts.set(id, updatedAttempts)}
+								/>
+							})}
+						</div>
+					</div>
+					<div className="save-error">
+						{ saveError != ''
+							? <p>{saveError}</p>
+							: null
+						}
+					</div>
+					<div className="button-holder">
+						<a tabIndex="1" className="cancel_button" onClick={onClose}>
+								Cancel
+						</a>
+						<a tabIndex="2" className="action_button green save_button" onClick={onSave}>
+							Save
+						</a>
+					</div>
+				</div>
+			</div>
+		</Modal>
+	)
+}
+
+export default ExtraAttemptsDialog
```

Comment:
new line at end of file (check editor settings)
```suggestion
export default ExtraAttemptsDialog

```